Index: bot/bot.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- bot/bot.py	(date 1540228146000)
+++ bot/bot.py	(date 1540229289299)
@@ -38,11 +38,13 @@
         self.stay_still_bonus = None
 
         self.game = hlt.Game()
-        self.callbacks = []
-        self.debug_maps = {}
         self.ships_targets: Dict[Ship, Tuple[int, int]] = {}
+        self.ships_to_home: Set[int] = set()
         self.max_ships_reached = 0
 
+        self.callbacks = []
+        self.debug_maps = {}
+
     def run(self):
         self.game.ready("BogdanDm")
         logging.info("Player ID: {}.".format(self.game.my_id))
@@ -100,9 +102,16 @@
         moves: List[Tuple[Ship, Iterable[Tuple[int, int]]]] = []
         for ship in my_ships:
             # Ship unloading
-            if ship.halite_amount >= constants.MAX_HALITE * self.ship_fill_k:
+            to_home = ship in self.ships_to_home
+            if to_home and ship.position == home:
+                logging.info(f"Ship#{ship.id} unload")
+                self.ships_to_home.remove(ship)
+                to_home = False
+
+            if to_home or ship.halite_amount >= constants.MAX_HALITE * self.ship_fill_k:
                 logging.info(f"Ship#{ship.id} moving home")
                 self.ships_targets[ship] = home
+                self.ships_to_home.add(ship)
                 moves.append((ship, list(self.game.map.get_unsafe_moves(ship.position, home))))
 
             else:
